src/garage/envs/dm_control/dm_control_env.py

```diff
@@ -47,34 +95,76 @@ def from_suite(cls, domain_name, task_name):
         Return:
             dm_control.suit.Task: the dm_control task environment
         """
-        return cls(suite.load(domain_name, task_name),
-                   name='{}.{}'.format(domain_name, task_name))
+        return cls(env=suite.load(domain_name, task_name),
+                   name='{}.{}'.format(domain_name, task_name),
+                   max_episode_length=max_episode_length)
+
+    def reset(self):
+        """Resets the environment.
+
+        Returns:
+            numpy.ndarray: The first observation. It must conform to
+            `observation_space`.
+            dict: The episode-level information. Note that this is not part
+            of `env_info` provided in `step()`. It contains information of
+            the entire episode， which could be needed to determine the first
+            action (e.g. in the case of goal-conditioned or MTRL.)
+
+        """
+        time_step = self._env.reset()
+        first_obs = flatten_observation(time_step.observation)['observations']
+
+        self._step_cnt = 0
+        self._last_observation = first_obs
+        # Populate episode_info if needed.
+        episode_info = {}
+        return first_obs, episode_info
 
     def step(self, action):
-        """Step the environment.
+        """Steps the environment with the action and returns a `EnvStep`.
 
         Args:
             action (object): input action
 
         Returns:
-            Step: The time step after applying this action.
-        """
-        time_step = self._env.step(action)
-        return Step(
-            flatten_observation(time_step.observation)['observations'],
-            time_step.reward, time_step.step_type == StepType.LAST,
-            **time_step.observation)
-
-    def reset(self):
-        """Reset the environment.
+            EnvStep: The time step resulting from the action.
 
-        Returns:
-            Step: The first time step.
+        Raises:
+            RuntimeError: if `step()` is called after the environment has been
+                constructed and `reset()` has not been called.
         """
-        time_step = self._env.reset()
-        return flatten_observation(time_step.observation)['observations']
+        if self._last_observation is None:
+            raise RuntimeError('reset() must be called before step()!')
+
+        dm_time_step = self._env.step(action)
+        if self._viewer:
+            self._viewer.render()
 
-    def render(self, mode='human'):
+        observation = flatten_observation(
+            dm_time_step.observation)['observations']
+
+        last_obs = self._last_observation
+        self._step_cnt += 1
+        self._last_observation = observation
+
+        # Determine step type
+        step_type = None
+        if dm_time_step.step_type == dm_StepType.MID:
+            step_type = StepType.TIMEOUT if \
```

Comment:
if you need to use a \ break, that's an indication that this the ternary operator probably isn't appropriate here